pages/creating_unifex_natives.md

```diff
@@ -0,0 +1,167 @@
+# Creating Unifex Natives
+
+## Introduction
+In this section we present how to create Unifex Natives. 
+We will show it by writing code that will be compiled both as NIF and CNode.
+
+## Preparation
+
+In order to start working, you need to prepare a few things:
+
+1. First, you have to add unifex to your dependencies as well as unifex and bundlex compilers.
+   Please refer to [Instalation](https://hexdocs.pm/unifex/readme.html#instalation) section to see how to do it. 
+2. After successful installation we should take a look at [Bundlex](https://github.com/membraneframework/bundlex).
+   Unifex uses Bundlex to compile the native code. 
+   You can think of Bundlex as a tool that generates build scripts responsible for including proper libs,  
+   compiling your native code and linking it with mentioned libs.
+   To make it work, create the `bundlex.exs` file in the project's root directory with the following content:
+    ```elixir
+    defmodule Example.BundlexProject do
+      use Bundlex.Project
+
+      def project() do
+        [
+          natives: natives(Bundlex.platform())
+        ]
+      end
+
+      def natives(_platform) do
+        [
+          example: [
+            deps: [unifex: :unifex],
+            src_base: "example",
+            sources: ["_generated/nif/example.c", "example.c"],
+            interface: :nif
+          ],
+          example: [
+           deps: [unifex: :unifex],
+           src_base: "example",
+           sources: ["_generated/cnode/example.c", "example.c"],
+           interface: :cnode
+          ] 
+        ]
+      end
+    end
+    ```
+   This defines two natives, both are called `example` that will be implemented in two `.c` files. 
+   The first one will be compiled as NIF and the second one as CNode. 
+   In the future bundlex is intended to do it by specifying only one native like:
+   ```elixir
+      example: [
+        deps: [unifex: :unifex],
+        src_base: "example",
+        sources: ["_generated/example.c", "example.c"],
+        interface: [:nif, :cnode]
+      ],
+   ```
+   but at this moment we have to do it in a little more descriptive way.
+   Source files have to be located in `c_src/example` directory.
+   More details on how to use bundlex can be found in its [documentation](https://hexdocs.pm/bundlex).
+
+## Native code
+
+Let's start by creating a `c_src/example` directory, and the files that will be needed:
+
+```bash
+mkdir -p c_src/example
+cd c_src/example
+touch example.c
+touch example.h
+touch example.spec.exs
+```
+
+You may wonder where is the `_generated/nif/example.c` and `_generated/cnode/example.c`. 
+Well, as the name suggests, it will be generated based on `example.spec.exs`!
+
+Here are the contents of `example.spec.exs`:
+
+```elixir
+module Example
```

Comment:
Not sure if this example is not too simple but on the other hand I linked some more advanced ones.